btw: I can see the warning locally with `CC=clang`, but it does not cause an error, although I should have `-Werror` also.

Via `CMAKE_EXTRA_FLAGS += "-DCMAKE_C_FLAGS=${CMAKE_C_FLAGS} -Werror"` in `local.mk`, and also tried `CMAKE_EXTRA_FLAGS += -DTRAVIS_CI_BUILD=ON` therein.

Adding `-Wall` to `"-DCMAKE_C_FLAGS` fixes it, but that should be enabled by default via `add_compile_options(-Wall …`.

```
[285/323] Building C object src/nvim/CMakeFiles/nvim.dir/tui/tui.c.o
../src/nvim/tui/tui.c:1610:48: warning: expression which evaluates to zero treated as a null pointer constant of type 'const char *' [-Wnon-literal-null-conversion]
        unibi_set_str(ut, unibi_set_lr_margin, '\0');
                                               ^~~~
1 warning generated.
```

Any idea?

--- a/src/nvim/tui/tui.c
+++ b/src/nvim/tui/tui.c
@@ -1607,7 +1607,7 @@ static void patch_terminfo_bugs(TUIData *data, const char *term,
       // Fix things advertised via TERM=xterm, for non-xterm.
       if (unibi_get_str(ut, unibi_set_lr_margin)) {
         ILOG("Disabling smglr with TERM=xterm for non-xterm.");
-        unibi_set_str(ut, unibi_set_lr_margin, '\0');
+        unibi_set_str(ut, unibi_set_lr_margin, (const char *)'\0');
       }
     }
 

--- a/src/nvim/tui/tui.c
+++ b/src/nvim/tui/tui.c
@@ -1607,7 +1607,7 @@ static void patch_terminfo_bugs(TUIData *data, const char *term,
       // Fix things advertised via TERM=xterm, for non-xterm.
       if (unibi_get_str(ut, unibi_set_lr_margin)) {
         ILOG("Disabling smglr with TERM=xterm for non-xterm.");
-        unibi_set_str(ut, unibi_set_lr_margin, (const char *)'\0');
+        unibi_set_str(ut, unibi_set_lr_margin, NULL);
       }
     }
 

